monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/ScanObservable.scala

```diff
@@ -40,24 +40,33 @@ class ScanObservable[A,R](
       source.unsafeSubscribeFn(
         new Subscriber[A] {
           implicit val scheduler = out.scheduler
-          private[this] var isDone = false
+          @volatile private[this] var isDone = false
           private[this] var state = initialState
+          private[this] val semaphore = AsyncSemaphore(1)
+          // Signal initial state immediately
+          private[this] var lastAck = out.onNext(state)
 
-          def onNext(elem: A): Future[Ack] = {
-            // Protects calls to user code from within the operator and
-            // stream the error downstream if it happens, but if the
-            // error happens because of calls to `onNext` or other
-            // protocol calls, then the behavior should be undefined.
-            var streamError = true
-            try {
-              state = f(state, elem)
-              streamError = false
-              out.onNext(state)
-            }
-            catch {
-              case NonFatal(ex) if streamError =>
-                onError(ex)
+          def onNext(elem: A): Future[Ack] = semaphore.greenLight { () =>
```

Comment:
You don't need a semaphore, because by the time `source.unsafeSubscribeFn` actually happens, the first `onNext(state)` was already sent, so you only need to take care of back-pressuring, therefore `lastAck.syncFlatMap` should be all that's necessary.